### PR TITLE
SE-9540 Fix for barcode blurriness

### PR DESCRIPTION
**Pull request by jmuscireum**

### CODE REVIEW

File: src/main/java/sirius/web/templates/ImageReplacedElementFactory.java [hunk 1]
```diff
@@ -61,6 +61,7 @@
     private static final String BARCODE_TYPE_EAN = "ean";
     private static final String BARCODE_TYPE_INTERLEAVED_2_OF_5 = "interleaved2of5";
     private static final String BARCODE_TYPE_INTERLEAVED_2_OF_5_CHECKSUMMED = "interleaved2of5checksummed";
+    private static final int BARCODE_SCALER = 5;
```
Comment (sabieber) [inline src/main/java/sirius/web/templates/ImageReplacedElementFactory.java:68]: BARCODE_SCALE_FACTOR

File: src/main/java/sirius/web/templates/ImageReplacedElementFactory.java [hunk 2]
```diff
@@ -180,6 +187,14 @@ private ReplacedElement createImageForBarcode(String type, String src, int cssWi
         return new ITextImageElement(fsImage);
     }
 
+    private int getBarcodeScaleFactor(int cssWidth, int cssHeight, java.awt.Image awtImage) {
+        if (cssWidth > 0) {
```
Comment (andyHa) [inline src/main/java/sirius/web/templates/ImageReplacedElementFactory.java:191]: use Math.max of both?

File: src/main/java/sirius/web/templates/ImageReplacedElementFactory.java [hunk 3]
```diff
@@ -180,6 +187,11 @@ private ReplacedElement createImageForBarcode(String type, String src, int cssWi
         return new ITextImageElement(fsImage);
     }
 
+    private int getBarcodeScaleFactor(int cssWidth, int cssHeight, java.awt.Image awtImage) {
```
Comment (sabieber) [inline src/main/java/sirius/web/templates/ImageReplacedElementFactory.java:190]:
```suggestion
    private int calculateBarcodeScaleFactor(int cssWidth, int cssHeight, java.awt.Image awtImage) {
```

File: src/main/java/sirius/web/templates/ImageReplacedElementFactory.java [hunk 4]
```diff
@@ -170,8 +170,15 @@ private ReplacedElement createImageForBarcode(String type, String src, int cssWi
         Barcode code = createBarcode(type);
         code.setCode(padCodeIfNecessary(code, src));
 
-        FSImage fsImage =
-                new ITextFSImage(Image.getInstance(code.createAwtImage(Color.BLACK, Color.WHITE), Color.WHITE));
+        java.awt.Image awtImage = code.createAwtImage(Color.BLACK, Color.WHITE);
+
+        int scaleFactor = getBarcodeScaleFactor(cssWidth, cssHeight, awtImage);
```
Comment (sabieber) [inline src/main/java/sirius/web/templates/ImageReplacedElementFactory.java:175]:
```suggestion
        int scaleFactor = calculateBarcodeScaleFactor(cssWidth, cssHeight, awtImage);
```

**Pull request merged**